dependencies.yaml

```diff
@@ -80,13 +80,35 @@ dependencies:
           - *cmake_ver
           - cython>=3.0.0
           - ninja
-          - rmm==24.2.*
       - output_types: conda
         packages:
           - scikit-build-core>=0.7.0
+          - rmm==24.2.*
       - output_types: [requirements, pyproject]
         packages:
           - scikit-build-core[pyproject]>=0.7.0
+      - output_types: [requirements, pyproject]
+        packages:
+          # pip recognizes the index as a global option for the requirements.txt file
+          # This index is needed for rmm.
+          - --extra-index-url=https://pypi.nvidia.com
+          - --extra-index-url=https://pypi.anaconda.org/rapidsai-wheels-nightly/simple
+          # we need --pre for rmm 24.2, as all current versions are pre-release (alpha)
+          - --pre
```

Comment:
We want to handle this with something like cudf's use of `alpha_spec` in a script: https://github.com/rapidsai/cudf/blob/6abef4a4746f1f9917711f372726023efdc21e85/ci/build_wheel.sh#L28-L35